src/Core/Resolvers/DWSqlQueryBuilder.cs

```diff
@@ -0,0 +1,205 @@
+            string predicates = JoinPredicateStrings(
+                                    structure.GetDbPolicyForOperation(EntityActionOperation.Read),
+                                    structure.FilterPredicates,
+                                    Build(structure.Predicates),
+                                    Build(structure.PaginationMetadata.PaginationPredicate));
+            string columns = WrappedColumns(structure);
+            string orderBy = $" ORDER BY {Build(structure.OrderByColumns)}";
+
+            string query = $"SELECT TOP {structure.Limit()} {columns}"
+                + $" FROM {fromSql}"
+                + $" WHERE {predicates}"
+                + orderBy;
+            return query;
+        }
+
+        /// <inheritdoc />
+        public string Build(SqlInsertStructure structure)
+        {
+            throw new NotImplementedException("DataWarehouse Sql currently does not support inserts");
+        }
+
+        /// <inheritdoc />
+        public string Build(SqlUpdateStructure structure)
+        {
+            throw new NotImplementedException("DataWarehouse sql currently does not support updates");
+        }
+
+        /// <inheritdoc />
+        public string Build(SqlDeleteStructure structure)
+        {
+            throw new NotImplementedException("DataWarehouse sql currently does not support deletes");
+        }
+
+        /// <inheritdoc />
+        public string Build(SqlExecuteStructure structure)
+        {
+            throw new NotImplementedException("DataWarehouse sql currently does not support executes");
+        }
+
+        /// <inheritdoc />
+        public string Build(SqlUpsertQueryStructure structure)
+        {
+            throw new NotImplementedException("DataWarehouse sql currently does not support updates");
+        }
+
+        /// <summary>
+        /// Add a JSON_QUERY wrapper on the column
+        /// </summary>
+        private string WrapSubqueryColumn(LabelledColumn column, SqlQueryStructure subquery)
+        {
+            string builtColumn = Build(column as Column);
+            if (subquery.IsListQuery)
+            {
+                return $"JSON_QUERY (COALESCE({builtColumn}, '[]'))";
+            }
+
+            return $"JSON_QUERY ({builtColumn})";
+        }
+
+        /// <summary>
+        /// Build columns and wrap columns which represent join subqueries
+        /// </summary>
+        private string WrappedColumns(SqlQueryStructure structure)
+        {
+            return string.Join(", ",
+                structure.Columns.Select(
+                    c => structure.IsSubqueryColumn(c) ?
+                        WrapSubqueryColumn(c, structure.JoinQueries[c.TableAlias!]) + $" AS {QuoteIdentifier(c.Label)}" :
+                        Build(c)
+            ));
+        }
+
+        /// <summary>
+        /// Builds the query to fetch result set details of stored-procedure.
```

Comment:
is this summary still valid for data warehouse?

Reply:
Stored procedures are supported in DW. This will be addressed in later pr's